src/io/flutter/run/coverage/FlutterCoverageEnabledConfiguration.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2021 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.run.coverage;
+
+import com.intellij.coverage.CoverageDataManager;
+import com.intellij.coverage.CoverageRunner;
+import com.intellij.execution.configurations.RunConfigurationBase;
+import com.intellij.execution.configurations.coverage.CoverageEnabledConfiguration;
+import com.intellij.openapi.application.ModalityState;
+import com.intellij.ui.GuiUtils;
+import io.flutter.FlutterBundle;
+import io.flutter.pub.PubRoot;
+import io.flutter.pub.PubRoots;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+public class FlutterCoverageEnabledConfiguration extends CoverageEnabledConfiguration {
+
+  public FlutterCoverageEnabledConfiguration(@NotNull RunConfigurationBase<?> configuration) {
+    super(configuration);
+    setCoverageRunner(CoverageRunner.getInstance(FlutterCoverageRunner.class));
+    final List<PubRoot> roots = PubRoots.forProject(configuration.getProject());
+    if (roots.isEmpty()) {
+      throw new RuntimeException(FlutterBundle.message("project.root.not.found"));
+    }
+    myCoverageFilePath = roots.get(0).getRoot().getCanonicalPath() + "/coverage/lcov.info";
+    GuiUtils.invokeLaterIfNeeded(
+      () -> setCurrentCoverageSuite(CoverageDataManager.getInstance(configuration.getProject()).addCoverageSuite(this)),
+      ModalityState.any());
+  }
+
+  public void setCoverageRunner(@Nullable final CoverageRunner coverageRunner) {
+    final String path = myCoverageFilePath;
```

Comment:
Why does `path` get set to `myCoverageFilePath` and then the latter is set to the former? Maybe a comment here would be helpful. Also confusing to see that `myCoverageFilePath` is set again later in the constructor.

Reply:
I'll add a comment explaining that the super method clears the value and we want to keep it. It seems like the constructor could just call the super method directly. This code went through quite a few iterations while trying to figure out how to make the Java coverage framework for us. Thanks for catching things that need cleaning up.